mesonbuild/compilers/d.py

```diff
@@ -341,14 +341,25 @@ def _translate_args_to_nongnu(self, args: T.List[str]) -> T.List[str]:
                 # Hence, we guess here whether the flag was intended to pass
                 # a linker search path.
 
+                if arg.startswith('-L='):
+                    suffix = arg[3:]
+                else:
+                    suffix = arg[2:]
+
+                # The user has passed a linker flag directly for the D compiler
+                if suffix.startswith('-L'):
+                    dcargs.append(arg)
+                    continue
```

Comment:
A user providing library search path to Meson should pass them in the form `-L=-L/some/path`, right?

Reply:
For D? yes.